docs/validate.rst

```diff
@@ -187,70 +187,31 @@ By default, no validation is enforced, but optionally, validation can be enabled
-
-    .. classmethod:: cls_checks(format, raises=())
-
-        Register a decorated function as *globally* validating a new format.
-
-        Any instance created after this function is called will pick up the supplied checker.
-
-        :argument str format: the format that the decorated function will check
-        :argument Exception raises: the exception(s) raised
-            by the decorated function when an invalid instance is
-            found. The exception object will be accessible as the
-            `jsonschema.exceptions.ValidationError.cause` attribute
-            of the resulting validation error.
 
-        .. deprecated:: v4.14.0
+Some formats require additional dependencies to be installed.
 
-            Use `FormatChecker.checks` on an instance instead.
+The easiest way to ensure you have what is needed is to install ``jsonschema`` using the ``format`` or ``format-nongpl`` extras.
 
+For example:
 
-.. autoexception:: FormatError
-    :noindex:
-    :members:
+    .. code-block:: sh
 
-
-There are a number of default checkers that `FormatChecker`\s know how
-to validate. Their names can be viewed by inspecting the
-`FormatChecker.checkers` attribute. Certain checkers will only be
-available if an appropriate package is available for use. The easiest way to
-ensure you have what is needed is to install ``jsonschema`` using the
-``format`` or ``format_nongpl`` collection of optional dependencies -- e.g.
-
-.. code-block:: sh
-
-   $ pip install jsonschema[format]
-
-which will install all of the below dependencies for all formats.
+        $ pip install jsonschema[format-nongpl]
 
 Or if you want to install MIT-license compatible dependencies only:
 
-.. code-block:: sh
+    .. code-block:: sh
 
-   $ pip install jsonschema[format_nongpl]
+        $ pip install jsonschema[format_nongpl]
 
-The non-GPL extra is intended to not install any direct dependencies
-that are GPL (but that of course end-users should do their own verification).
-At the moment, it supports all the available checkers except for ``iri`` and
-``iri-reference``.
+The non-GPL extra is intended to not install any direct dependencies that are GPL licensed (but that of course end-users should do their own verification).
+At the moment, it supports all the available checkers except for ``iri`` and ``iri-reference``.
 
-The more specific list of available checkers, along with their requirement
-(if any,) are listed below.
+The more specific list of formats along with any additional dependencies they have is shown below.
 
-.. note::
+.. warning::
 
-    If the following packages are not installed when using a checker
-    that requires it, validation will succeed without throwing an error,
-    as specified by the JSON Schema specification.
+    If a dependency is not installed when using a checker that requires it, validation will succeed without throwing an error, as also specified by the specification.
```

Comment:
This is so terrible. Can't you add an unofficial strict mode that fails when

1. an format is added without a format checker
2. and unknown format is specified
3. an format is specified that you lack the deps for?